compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -391,6 +391,21 @@ impl<'a> LlvmArchiveBuilder<'a> {
             ret
         }
     }
+
+    fn i686_decorated_name(import: &DllImport) -> CString {
+        let name = import.name.to_string();
```

Comment:
```suggestion
        let name = import.name;
```
Symbols can be `format!`ed without allocating a new string.